From 11829635f6a9194f1173798f6afa420bbd643b68 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 3 Jan 2017 17:39:25 -0500
Subject: [PATCH 2/9] Trivial fix for an error message in
 get_header_from_pointer()

get_header_from_pointer() prints an error when a double-free happens,
but it doesn't have a newline.  It obviously should.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 grub-core/kern/mm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grub-core/kern/mm.c b/grub-core/kern/mm.c
index ee88ff61187a..9c73c12f1be7 100644
--- a/grub-core/kern/mm.c
+++ b/grub-core/kern/mm.c
@@ -99,7 +99,7 @@ get_header_from_pointer (void *ptr, grub_mm_header_t *p, grub_mm_region_t *r)
 
   *p = (grub_mm_header_t) ptr - 1;
   if ((*p)->magic == GRUB_MM_FREE_MAGIC)
-    grub_fatal ("double free at %p", *p);
+    grub_fatal ("double free at %p\n", *p);
   if ((*p)->magic != GRUB_MM_ALLOC_MAGIC)
     grub_fatal ("alloc magic is broken at %p: %lx", *p,
 		(unsigned long) (*p)->magic);
-- 
2.14.3

